openquake/risklib/workflows.py

```diff
@@ -141,19 +145,35 @@ def value(self, loss_type, time_event=None):
         """
         if loss_type == 'fatalities':
             return self.values['fatalities_' + str(time_event)]
-        return self.calc(loss_type, self.values, self.area, self.number)
+        try:
+            val = self._cost[loss_type]
+        except KeyError:
+            val = self.calc(loss_type, self.values, self.area, self.number)
+            self._cost[loss_type] = val
+        return val
 
     def deductible(self, loss_type):
         """
-        :returns: the deductible of the asset for `loss_type`
+        :returns: the deductible fraction of the asset cost for `loss_type`
         """
-        return self.deductibles[loss_type]
+        val = self.calc(loss_type, self.deductibles, self.area, self.number)
+        if self.calc.deduct_abs:  # convert to relative value
+            return val / self.calc(loss_type, self.values,
+                                   self.area, self.number)
+        else:
+            return val
 
     def insurance_limit(self, loss_type):
```

Comment:
insurance limit and deductible are almost identical, but I guess we can keep them as they are